plot_equality_data.R

```diff
@@ -0,0 +1,280 @@
+          filelist2 = lapply(ssp_plot, namefun, y=y, z=z, timeframe=timeframe) 
+          filelist=c(filelist, filelist2)
+        }
+      }
+    }
+  } 
+  for (y in variable_rcp){
+    for (z in variable_risk){
+      for (timeframe in variable_timeframe){
+        filelist2 = lapply(ssp_plot, namefun_nocut, y=y, z=z, timeframe=timeframe) 
+        filelist=c(filelist, filelist2)
+      }
+    }
+  }
+}
+
+if (grepl("dice", dmg_f)){
+  for (y in variable_rcp){
+    for (z in variable_risk){
+      filelist2 = lapply(ssp_plot, namefun_dice, y=y, z=z)
+      filelist=c(filelist, filelist2)
+    }
+  }
+}  
+
+
+results_table <- data.table(ssp=integer(), rcp=numeric(), eta=numeric(), PRTP=numeric(), 
+                            damages=character(), indicator=character(), value=numeric())
+columns_to_save = c("mean")
+origin = getwd()
+compare_results = results_table
+for (file in filelist) {
+  load(paste0(origin, file))
+  sspnum = as.numeric(substr(strsplit(file, split = "SSP")[[1]][2], 1, 1))
+  rcpnum = substr(strsplit(file, split = "rcp")[[1]][2], 1, 2)
+  rcpnum = sub("(.{1})(.*)", "\\1.\\2", rcpnum)
+  damages = if(grepl("djo", file)) "Dell" else if(grepl("dice", file)) "Dice" else (
+    if (grepl("horizon2100", file)) "Burke 2100" else "Burke 2200")
+  if (grepl("_30C", file)){
+    damages = str_c(damages, " 30C")
+  }
+  for (column in columns_to_save){
+    if(type_str == "eri_eq_statscc_2020d"){
+      use_table = eri_eq_stat_wscc
+      # We can also compare the impact of inequality aversion
+      compare_file = str_replace(file, "eri_eq_statscc_2020d", "statscc_")
+      load(paste0(origin, compare_file))
+      compare_table = stat_scc
+    } else {
+      use_table = poor_prefer_10
+    }
+    rows_to_save = lapply(use_table$ID, strsplit, split="_")
+    rowind = 0
+    for (row in rows_to_save){
+      rowind = rowind + 1
+      if (row[[1]][1] %in% c("NA", "3") ) next
+      PRTPnum = row[[1]][1]
+      etanum = row[[1]][2]
+      results_table <- rbind(results_table, list(
+        sspnum, rcpnum, etanum, PRTPnum, damages, column, use_table[[rowind, column]])
+      )
+      if(type_str == "eri_eq_statscc_2020d"){
+        compare_results <- rbind(
+          compare_results, list(sspnum, rcpnum, etanum, PRTPnum, damages, column, compare_table[[
+            which(compare_table$ID == paste(PRTPnum, etanum, "NA", "WLD", sep="_")), column
+          ]])
+        )
+      }
+    }
+  }
+}
+if (type_str == "poor_pref_10dollars"){
+  compare_results$value = compare_results$value * dollar_val_2020  
+}
+if (grepl("nice", dmg_f)){
+  nice_data <- read.csv("./data/nice_imports/interpolated_scc_scenarios_v4_2025.csv")
+  nice_data$ssp = substr(nice_data$Scenario, 4, 4)
+  nice_data$rcp = sapply(nice_data$Scenario, function(x) strsplit(x, "-")[[1]][2])
+  nice_data$rcp[nice_data$rcp=="Baseline"] = "85"
+  nice_data_poor = nice_data[nice_data$whose=="poorest", ]
+  len_nice = nrow(nice_data_poor)
+  nice_results_table <- data.table(
```

Comment:
I get a warning message that columns have 0 rows so are filled with NA. Could we get rid of the message, because the code does work